### DISCUSSION TITLE
Is there a mistake in which formula to use when buying certain amount of X tokens ?

### DISCUSSION THREAD

**Discussion by falcon-lin-development** [Jeiwan/uniswapv3-book#100] [Milestone 0. Introduction] [answered]
<img width="781" alt="Screenshot 2024-06-15 at 12 50 29 PM" src="https://github.com/Jeiwan/uniswapv3-book/assets/50874568/86c8b241-f5c6-4a9a-a467-119b0da07e9d">

this says 
```
we can always find the input amount using the Δx formula (when we want to buy a known amount of tokens).
```

But in fact, what we need to do is to switch place of x and y and still use the Delta(y) formula? 

Reason: 
1) we need to take account into the transaction fee.
2) Buying Token X is essentially selling the Token Y.

Please correct me if I'm wrong

**Comment by 0xkaranchauhan** [comment, marked as answer]
@falcon-lin-development 
No need to switch the variables x and y and use the Δy formula. Instead, use the provided formulas as intended:

- To find the output amount of token y when selling a known amount of token x, use **Δy = (yrΔx) / (x+rΔx)**.
- To find the input amount of token x needed to buy a known amount of token y, use **Δx = (xΔy) / (r(y−Δy))**.
These formulas already incorporate the transaction fees and respect the relationships between supply and demand.

**Reply by falcon-lin-development** [reply]
I see it now. Thanks a lot.
